### PR TITLE
[fuchsia] Agis handshake on Fuchsia

### PR DESCRIPTION
**Pull request by rosasco-wk**
- Add elements of establishing a socket connection on Fuchsia.
- Update Fuchsia build to latest rules in Bazel SDK (sdk-integration).
- Revise connection abstraction to make usable for Fuchsia

### CODE REVIEW

**Review by yalcinmelihyasin — APPROVED** [review]
LGTM. 

But please confirm before merging.

**Comment by rosasco-wk** [comment]
Will confirm

File: WORKSPACE [hunk 1]
```diff
@@ -35,6 +35,8 @@ load("@local_config_fuchsia//:workspace.bzl", "fuchsia_base_dependencies")
 fuchsia_base_dependencies(locals = LOCALS)
 load("@local_config_fuchsia//:fuchsia_sdk.bzl", "fuchsia_sdk_dependencies")
 fuchsia_sdk_dependencies(locals = LOCALS)
+load("@fuchsia_clang//:defs.bzl", "register_clang_toolchains")
```
Comment (chaselatta) [inline WORKSPACE:38]: load("@local_config_fuchsia//:fuchsia_clang.bzl", "fuchsia_clang_dependencies")
fuchsia_clang_dependencies(locals = LOCALS)

//fuchsia_clang.bzl
load("@gapid//tools/build/rules:repository.bzl", "maybe_repository")
load("@fuchsia_clang//:defs.bzl", "register_clang_toolchains")

def fuchsia_clang_dependencies(locals = {}):
  maybe_repository(
    fuchsia_clang_repository,
    name = "fuchsia_clang",
    locals = locals,
    sdk_root_label = "@fuchsia_sdk",
  )
  register_clang_toolchains()

Comment (rosasco-wk) [reply WORKSPACE]: Thanks Chase.  This will be on a different PR.

**Pull request merged**